core/src/main/java/org/elasticsearch/index/translog/TruncateTranslogCommand.java

```diff
@@ -0,0 +1,217 @@
+                        translogGeneration, translogUUID);
+            }
+            terminal.println("Translog Generation: " + translogGeneration);
+            terminal.println("Translog UUID      : " + translogUUID);
+
+            Path tempEmptyCheckpoint = translogPath.resolve("temp-" + Translog.CHECKPOINT_FILE_NAME);
+            Path realEmptyCheckpoint = translogPath.resolve(Translog.CHECKPOINT_FILE_NAME);
+            Path tempEmptyTranslog = translogPath.resolve("temp-" + Translog.TRANSLOG_FILE_PREFIX +
+                            translogGeneration + Translog.TRANSLOG_FILE_SUFFIX);
+            Path realEmptyTranslog = translogPath.resolve(Translog.TRANSLOG_FILE_PREFIX + translogGeneration + Translog.TRANSLOG_FILE_SUFFIX);
+
+            // Write empty checkpoint and translog to empty files
+            long gen = Long.parseLong(translogGeneration);
+            writeEmptyCheckpoint(tempEmptyCheckpoint, gen);
+            writeEmptyTranslog(tempEmptyTranslog, translogUUID);
+
+            terminal.println("Removing existing translog files");
+            IOUtils.rm(translogFiles.toArray(new Path[]{}));
+
+            terminal.println("Creating new empty checkpoint at [" + realEmptyCheckpoint + "]");
+            Files.move(tempEmptyCheckpoint, realEmptyCheckpoint, StandardCopyOption.ATOMIC_MOVE);
+            terminal.println("Creating new empty translog at [" + realEmptyTranslog + "]");
+            Files.move(tempEmptyTranslog, realEmptyTranslog, StandardCopyOption.ATOMIC_MOVE);
+
+        } catch (LockObtainFailedException lofe) {
+            throw new ElasticsearchException("Failed to lock shard's directory at [" + idxLocation + "], is Elasticsearch still running?");
+        }
+
+        terminal.println("Done.");
+    }
+
+    /** Coerce a long into a byte array */
+    private static byte[] getLongBytes(long val) {
+        return new byte[] { (byte) (val >> 56), (byte) (val >> 48), (byte) (val >> 40), (byte) (val >> 32),
+                            (byte) (val >> 24), (byte) (val >> 16), (byte) (val >> 8), (byte) val };
+    }
+
+    /** Write a checkpoint file to the given location with the given generation */
+    public static void writeEmptyCheckpoint(Path filename, long translogGeneration) throws IOException {
+        try (OutputStreamDataOutput out = new OutputStreamDataOutput(new FileOutputStream(filename.toFile()))) {
+            Checkpoint emptyCheckpoint = new Checkpoint(43, 0, translogGeneration);
+            emptyCheckpoint.write(out);
+        }
+    }
+
+    /** Write a translog containing the given translog UUID to the given location */
+    public static void writeEmptyTranslog(Path filename, String translogUUID) throws IOException {
+        try (FileOutputStream out = new FileOutputStream(filename.toFile())) {
+            out.write(EMPTY_TRANSLOG_PREFIX);
+            out.write(translogUUID.getBytes());
```

Comment:
I think you should fsync after writing these bytes, e.g. `out.getChannel().force(true)`, like the translog code does?  Would be horribly unlucky if the user ran this tool then their box crashed and it left the translog corrupt.